File: EJERCICIO 1/C++/RegistroEstudiantes.cpp
#include <iostream>
#include <fstream>   // Para manejo de archivos
#include <string>
using namespace std;

// Clase que representa a un Estudiante
class Estudiante {
private:
    // Atributos de la clase
    string cedula;
    string nombre;
    string carrera;
    double promedio;

public:
    // Constructor: inicializa todos los atributos
    Estudiante(string cedula, string nombre, string carrera, double promedio) {
        this->cedula   = cedula;
        this->nombre   = nombre;
        this->carrera  = carrera;
        this->promedio = promedio;
    }

    // Método que guarda los datos del estudiante en el archivo
    void guardarEnArchivo() {
        // ios::app activa el modo append: no sobreescribe registros anteriores
        ofstream archivo("estudiantes.txt", ios::app);

        if (archivo.is_open()) {
            // Escribir los datos separados por '|' y terminar con salto de línea
            archivo << cedula << "|" << nombre << "|"
                    << carrera << "|" << promedio << endl;
            archivo.close(); // Cerrar el flujo de escritura
            cout << "Estudiante registrado exitosamente en estudiantes.txt" << endl;
        } else {
            // Manejo de error si no se puede abrir/crear el archivo
            cout << "Error: no se pudo abrir el archivo." << endl;
        }
    }
};

// Función principal
int main() {
    string cedula, nombre, carrera;
    double promedio;

    // Solicitar y leer cada dato del estudiante
    cout << "Ingrese la cedula del estudiante: ";
    getline(cin, cedula);

    cout << "Ingrese el nombre del estudiante: ";
    getline(cin, nombre);

    cout << "Ingrese la carrera del estudiante: ";
    getline(cin, carrera);

    cout << "Ingrese el promedio del estudiante: ";
    cin >> promedio;

    // Crear objeto Estudiante con los datos ingresados
    Estudiante est(cedula, nombre, carrera, promedio);

    // Llamar al método que guarda en el archivo
    est.guardarEnArchivo();

    return 0;
}

File: EJERCICIO 1/Java/RegistroEstudiantes.java
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

// Clase que representa a un Estudiante
class Estudiante {
    // Atributos de la clase
    private String cedula;
    private String nombre;
    private String carrera;
    private double promedio;

    // Constructor: inicializa todos los atributos
    public Estudiante(String cedula, String nombre, String carrera, double promedio) {
        this.cedula  = cedula;
        this.nombre  = nombre;
        this.carrera = carrera;
        this.promedio = promedio;
    }

    // Método que guarda los datos del estudiante en el archivo
    public void guardarEnArchivo() {
        // 'true' activa el modo append: no sobreescribe registros anteriores
        try (BufferedWriter bw = new BufferedWriter(new FileWriter("estudiantes.txt", true))) {
            // Formatear la línea con separador '|'
            String linea = cedula + "|" + nombre + "|" + carrera + "|" + promedio;
            bw.write(linea);   // Escribir los datos
            bw.newLine();      // Agregar salto de línea al final del registro
            System.out.println("Estudiante registrado exitosamente en estudiantes.txt");
        } catch (IOException e) {
            // Manejo de error si no se puede escribir el archivo
            System.out.println("Error al escribir el archivo: " + e.getMessage());
        }
    }
}

// Clase principal con el método main
public class Main {
    public static void main(String[] args) {
        // Crear objeto Scanner para leer datos del teclado
        Scanner sc = new Scanner(System.in);

        // Solicitar y leer cada dato del estudiante
        System.out.print("Ingrese la cédula del estudiante: ");
        String cedula = sc.nextLine();

        System.out.print("Ingrese el nombre del estudiante: ");
        String nombre = sc.nextLine();

        System.out.print("Ingrese la carrera del estudiante: ");
        String carrera = sc.nextLine();

        System.out.print("Ingrese el promedio del estudiante: ");
        double promedio = sc.nextDouble();

        // Crear objeto Estudiante con los datos ingresados
        Estudiante est = new Estudiante(cedula, nombre, carrera, promedio);

        // Llamar al método que guarda en el archivo
        est.guardarEnArchivo();

        sc.close(); // Cerrar el Scanner
    }
}
